app/mailers/user_mailer.rb

```diff
@@ -16,25 +16,38 @@ def reservation_status_update(reservation, receipt = false) # rubocop:disable al
     @reservation = reservation
     @status = @reservation.status
 
-    return if !receipt && @status == 'returned overdue' &&
+    return if !receipt && @status == 'returned' && @reservation.overdue &&
               @reservation.equipment_model.late_fee == 0
 
     return if receipt && @reservation.checked_out.nil?
 
+    if @reservation.overdue
+      if @status == 'returned'
+        @status = 'returned overdue'
+      else
+        @status = 'overdue'
+      end
+    end
+
```

Comment:
All the method would need to do would be concatenate the status, overdue bool, and status flags into a readable string. It wouldn't be anywhere close to the madness of the old status method that compared dates, etc.